File: solutions/Java/reverse-string/5/src/main/java/ReverseString.java
import java.text.Normalizer;

class ReverseString {

    String reverse(String inputString) {
        String normalizedString = Normalizer.normalize(inputString, Normalizer.Form.NFC);
        StringBuilder reversedStr = new StringBuilder(normalizedString);
        return reversedStr.reverse().toString();
    }

    public static void main(String[] args) {
        ReverseString reverser = new ReverseString();
        String input = "Amélie Gruyères";
        String reversed = reverser.reverse(input);
        System.out.println(reversed);
    }
}

File: solutions/C/reverse-string/2/reverse_string.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "reverse_string.h"

char* reverse(char *str) {
    int length = strlen(str);
    char *reversed = malloc(length + 1); 

    for (int i = 0; i < length; i++) {
        reversed[i] = str[length - 1 - i];
    }

    reversed[length] = '\0'; 
    return reversed;
}


